.github/workflows/pipeline.yml

```diff
@@ -1,20 +1,40 @@
 ---
 name: Pipeline
+
 on: [push]
+
 jobs:
   tests:
-    name: Unit Tests
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-22.04
+
     strategy:
       fail-fast: false
       matrix:
-        ruby: [2.5, 2.6, 2.7, '3.0', 3.1]
+        ruby: [2.5, 2.6, 2.7, '3.0', 3.1, 3.2]
+        gemfile:
+          - Gemfile
+          - gemfiles/activesupport_5.gemfile
+          - gemfiles/activesupport_6.gemfile
+          - gemfiles/activesupport_7.gemfile
+        exclude:
+          - ruby: 2.5
+            gemfile: gemfiles/activesupport_7.gemfile
+          - ruby: 2.6
+            gemfile: gemfiles/activesupport_7.gemfile
+
+    env:
+      BUNDLE_GEMFILE: ${{ matrix.gemfile }}
+
+    name: Unit Tests (${{ matrix.ruby }} - ${{ matrix.gemfile }})
+
     steps:
-    - uses: actions/checkout@v2
-    - uses: ruby/setup-ruby@v1
-      with:
-        ruby-version: ${{ matrix.ruby }}
-        bundler: 2.2.29
-        bundler-cache: true
-    - name: Unit Tests
-      run: bundle exec rspec
+      - name: Check out
```

Comment:
🔧 We don’t require names for steps that are self evident by the action that they are using. Could we set this back?

Reply:
Although we may be a bit succinent in defining actions that are self-evident, I believe it doesn't hurt to be a bit explicit especially when it comes to yaml files that are harder to read/understand the flow. I think this is one of those areas where readability trumps conciseness. Let me know your thoughts.